src/modules/dashboard/components/ActionsPage/ActionsPage.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useEffect, useMemo } from 'react';
+import { useParams, Redirect } from 'react-router-dom';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import Heading from '~core/Heading';
+import TextDecorator from '~lib/TextDecorator';
+import UserMention from '~core/UserMention';
+import LoadingTemplate from '~pages/LoadingTemplate';
+import Button from '~core/Button';
+import CopyableAddress from '~core/CopyableAddress';
+
+import TransactionHash, { Hash } from './TransactionHash';
+
+import NakedMoleImage from '../../../../img/naked-mole.svg';
+
+import {
+  useTransactionLazyQuery,
+  useUserLazyQuery,
+  useColonyFromNameQuery,
+  useUser,
+} from '~data/index';
+import { isTransactionFormat } from '~utils/web3';
+import { STATUS } from './types';
+import { NOT_FOUND_ROUTE } from '~routes/index';
+
+import styles from './ActionsPage.css';
+
+const MSG = defineMessages({
+  actionTitle: {
+    id: 'dashboard.ActionsPage.actionTitle',
+    defaultMessage: `{name, select,
+      false {Unknown Transaction}
+      other {{name}}
+    } {user, select,
+      false {}
+      other {by {user}}
+    }`,
+  },
+  loading: {
+    id: 'dashboard.ActionsPage.loading',
+    defaultMessage: `Loading Transaction`,
+  },
+  transactionNotFound: {
+    id: 'dashboard.ActionsPage.transactionNotFound',
+    defaultMessage: `Ooops, transaction not found`,
+  },
+  returnToColony: {
+    id: 'dashboard.ActionsPage.returnToColony',
+    defaultMessage: `Return to colony`,
+  },
+});
+
+/**
+ * @NOTE On the specific colony address type
+ *
+ * This came about as a result of hooking into the result of the colony query,
+ * on the client side query, before it sends the result on to the server query,
+ * and act upon that if that's in an error state (in which case, it won't actually
+ * reach the server)
+ *
+ * See the comment below, where we actually set the reverseENSAddress for a more
+ * in depth explanation.
+ */
+type SuperSpecificColonyAddress = string | Error;
+
+const displayName = 'dashboard.ActionsPage';
+
+/*
+ * @TODO We need a decent way of detecting pending transaction
+ *
+ * Currently we only get these two states back from the chanin, and even so
+ * we only get it for post-byzantium forks
+ *
+ * I think there's a way of looking at the value of `blockNumber` but I need
+ * to look into it further
+ */
```

Comment:
I think I have a way to do this. 

Damn, this PR just keeps dragging me back ...

Reply:
I've implemented the pending state via dfdd3292ba4cde86f2750ea200fc03a7ec0a28d7